cubedash/_pages.py

```diff
@@ -273,10 +274,34 @@ def _get_grouped_products() -> List[Tuple[str, List[ProductWithSummary]]]:
     # Which field should we use when grouping products in the top menu?
     group_by_field = app.config.get("CUBEDASH_PRODUCT_GROUP_BY_FIELD", "product_type")
     group_field_size = app.config.get("CUBEDASH_PRODUCT_GROUP_SIZE", 5)
+    group_by_regex = app.config.get("CUBEDASH_PRODUCT_GROUP_BY_REGEX", None)
 
-    # Group using the configured key, or fall back to the product name.
-    def key(t):
-        return t[0].fields.get(group_by_field) or t[0].name
+    if group_by_regex:
+        try:
+            regex_group = {}
+            for regex, group in group_by_regex:
+                regex_group[re.compile(regex.strip())] = group.strip()
```

Comment:
It wont really hurt in this case (as product names can't have whitespace), but I'd rather not strip regexps when they're specified as a precise string literals. If we do want this sort of behaviour I'd rather go all the way and use the [verbose](https://docs.python.org/3/howto/regex.html#compilation-flags) flag.... but I'm not sure.